docs/content/storage-drivers/s3.md

```diff
@@ -59,6 +59,10 @@ Amazon S3 or S3 compatible services for object storage.
 
 `loglevel`: (optional) Valid values are: `off` (default), `debug`, `debugwithsigning`, `debugwithhttpbody`, `debugwithrequestretries`, `debugwithrequesterrors` and `debugwitheventstreambody`. See the [AWS SDK for Go API reference](https://docs.aws.amazon.com/sdk-for-go/api/aws/#LogLevelType) for details.
 
+**NOTE:** Currently the S3 storage driver does not support S3 API compatible storage that
+does not allow combining the last part in the multipart upload into a part that is bigger
+than the preconfigured `chunkSize`.
```

Comment:
For the upload to be resumable, the storage provider needs to allow uploading more parts following the oversized one, such that the oversized part is not the last part. The driver is incompatible with any storage provider which requires all parts except the last one to be the same size, irrespective of whether the last part could be bigger.

```suggestion
**NOTE:** Currently the S3 storage driver only supports S3 API compatible storage that
allows parts of a multipart upload to vary in size. [Cloudflare R2 is not supported.](https://developers.cloudflare.com/r2/objects/multipart-objects/#limitations)
```